CMakeLists.txt

```diff
@@ -477,6 +479,7 @@ include(cmake/install.cmake)
 # to enable tests. It creates a "BUILD_TESTING" option, and runs
 # `enable_testing()`. See https://cmake.org/cmake/help/latest/module/CTest.html
 if(SINGULARITY_BUILD_TESTS)
+  include(ExternalProject) # for downloading files
```

Comment:
Could this be moved to `test/CMakeLists.txt`?

Reply:
This is a snippet from a flecsi code I worth with

```cmake
if(ENABLE_UNIT_TESTS)
  include(FetchContent)
  FetchContent_Declare(matrix494_bus
	URL https://www.cise.ufl.edu/research/sparse/MM/HB/494_bus.tar.gz
	URL_HASH SHA256=9b2d33c6fe52ee5b4980da1a2d2b78201924a36714234757677fa5c829c483ca)
# ...
FetchContent_MakeAvailable(matrix494_bus)

# can use ${matrix494_bus_SOURCE_DIR}/data.dat, or whatever

endif()

```

Reply:
yes I'll move this. And I can switch to `FetchContent`. I just tried something until it worked. ;)